File: WEEKLYCODE#3/divisible_sum_pairs/Java/divisible_sum_pairs.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution
{
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
            int n = input.nextInt();
            int k = input.nextInt();
            int counter = 0,i,j;
            int a[] = new int[n];
            //taking input
            for ( i = 0; i < n; i++) 
                a[i] = input.nextInt();
            //checking the condition
            for(i = 0; i < n; i += 1)
                for(j = i + 1; j < n; j += 1)
                    if((a[i]+a[j])%k==0)
                        counter+=1;
            //printing result
            System.out.println(counter);
    }
}

File: WEEKLYCODE#3/divisible_sum_pairs/C/divisible_sum_pairs.c
#include<stdio.h>
int main(){
    int n,k,counter=0,i,j;
    //taking inputs
    scanf("%d%d",&n,&k);
    int ar[n];
    for(i=0;i<n;i++)
        scanf("%d",&ar[i]);
    //checking pairs
    for(i=0;i<n;i++)
        for(j=i+1;j<n;j++)
            if((ar[i]+ar[j])%k==0)
                counter+=1;
    //printing result
    printf("%d",counter);
    return 0;
}
